pkg/codegen/dotnet/gen.go

```diff
@@ -114,11 +115,17 @@ func namespaceName(namespaces map[string]string, name string) string {
 	if ns, ok := namespaces[name]; ok {
 		return ns
 	}
-	names := strings.Split(name, "-")
-	for i, name := range names {
-		names[i] = Title(name)
+
+	// name could be a qualified module name so first split on /
+	parts := strings.Split(name, tokens.QNameDelimiter)
+	for i, part := range parts {
+		names := strings.Split(part, "-")
+		for i, name := range names {
+			names[i] = Title(name)
```

Comment:
```suggestion
		for j, name := range names {
			names[j] = Title(name)
```

Maybe?  I know it's correct either way, but I did stumble over this wondering if it was correct for a second.  Having separate names instead of shadowing the variable here feels a little clearer to me.